tiledb/cloud/_results/results.py

```diff
@@ -46,14 +43,27 @@ def wrap(cls, func: Callable[..., _T]) -> Callable[..., Result[_T]]:
         return lambda *args, **kwargs: cls(func(*args, **kwargs))
 
 
-@dataclasses.dataclass()
+_SENTINEL = object()
+
+
+# Not frozen, but externally immutable, and callers should *treat* it like it's
+# frozen.
+@attr.s()
 class RemoteResult(Result[_T], Generic[_T]):
     """A response from running a UDF remotely."""
 
     def get(self) -> _T:
         """Decodes the response from the server."""
         try:
-            return self.decoder.decode(self.body)
+            with self._result_lock:
+                if self._decoded is _SENTINEL:
+                    if self._body is not None:
+                        self._decoded = self.decoder.decode(self._body)
```

Comment:
Do we need to keep `_body` after decoding?

Reply:
We do not! Good idea.